usb/esp_tinyusb/tusb_cdc_acm.c

```diff
@@ -380,28 +380,25 @@ esp_err_t tusb_cdc_acm_init(const tinyusb_config_cdcacm_t *cfg)
     /* Buffers */
 
     acm->ringbuf_read_mux = xSemaphoreCreateMutex();
-    if (acm->ringbuf_read_mux == NULL) {
-        ESP_LOGE(TAG, "Creation of a ringbuf mutex failed");
-        free_obj(itf);
-        return ESP_ERR_NO_MEM;
-    }
+    ESP_GOTO_ON_FALSE(acm->ringbuf_read_mux, ESP_ERR_NO_MEM, clean, TAG, "Creation of a ringbuf mutex failed");
 
     acm->rx_tfbuf = malloc(CONFIG_TINYUSB_CDC_RX_BUFSIZE);
-    if (!acm->rx_tfbuf) {
-        ESP_LOGE(TAG, "Creation buffer error");
-        free_obj(itf);
-        return ESP_ERR_NO_MEM;
-    }
+    ESP_GOTO_ON_FALSE(acm->rx_tfbuf, ESP_ERR_NO_MEM, clean, TAG, "Creation buffer error");
+
     acm->rx_unread_buf_sz = cfg->rx_unread_buf_sz == 0 ? RX_UNREADBUF_SZ_DEFAULT : cfg->rx_unread_buf_sz;
     acm->rx_unread_buf = xRingbufferCreate(acm->rx_unread_buf_sz, RINGBUF_TYPE_BYTEBUF);
-    if (acm->rx_unread_buf == NULL) {
-        ESP_LOGE(TAG, "Creation buffer error");
-        free_obj(itf);
-        return ESP_ERR_NO_MEM;
-    } else {
-        ESP_LOGD(TAG, "Comm Initialized buff:%d bytes", cfg->rx_unread_buf_sz);
-        return ESP_OK;
+    ESP_GOTO_ON_FALSE(acm->rx_unread_buf, ESP_ERR_NO_MEM, clean, TAG, "Creation buffer error");
+
+    ESP_LOGD(TAG, "Comm Initialized buff:%d bytes", cfg->rx_unread_buf_sz);
+    return ESP_OK;
+clean:
+    if (acm->rx_tfbuf) {
```

Comment:
Hi Saurabh, 
Great job!

One thing that I want to clarify for me is the next. During the tusb_cdc_acm_init function we:
- init tinyusb class device (with tinyusb_cdc_init())
- allocate cdc acm object
- create mutex (ringbuf_read_mux)
- allocate rx_tfbuf 
- allocate ring buffer (rx_unread_buf). 

During my attempts to understand the logic I see, that two main things should be achieved at the end:
1. We can use label goto when we need to separate some different steps from each other.
2. We should free all object, that was allocated at the step/steps before. 

In this case I am concern about the logic, when we need to use only one label (As much as much often at this time we can avoid using that at all). 
The second one tells us that we need to get freed all object had been allocated. That means that we need to do it in a backwards order: 
- free ring buffer
- free rx buffer
- delete mutex
- free cdc acm object
- deinit class usb device

From that point of view, it looks that with the goto labels we can do the following: 
`

    init_tinyusb_class_device();
        if not goto init_fail;
    allocate_cdc_acm_object();
        if not goto allocate_object_fail;
    create_mutex();
        if not goto create_mutex_failure;
    allocate_rx_tfbuf();
        if not goto alloc_buf_err;
    allocate_ring_buffer();
         if not goto ring_buffer_fail;
    return OK;

    ring_buffer_fail:
         free_rx_buffer();
    alloc_buf_error:
         delete_mutex();
    create_mutex_failure:
        free_cdc_acm_object();
    allocate_object_fail:
        deinit_class_usb_device();
    init_fail: 
        // nothing else to free
        return ERR;
`

Honestly, I am not a fan of the goto labels, but in such case it seems to be at least justified. 
Maybe I not understand the logic or I do not have the whole image...But it looks that in the suggested logic we: 
- use goto labels for their purpose
- freed all the object had been allocated... 

That is the only thing I would like to highlight.

Reply:
C standard says that passing NULL to `free` does nothing. So in order to have the clean-up procedure as trivial as possible, we usually use only one label where we free everything (even if the allocation did not occur). Does that make sense to you @roma-jam ?

Very good answer for this question, for you reference: https://stackoverflow.com/questions/6084218/is-it-good-practice-to-free-a-null-pointer-in-c

Reply:
@tore-espressif If the clean-up procedure is only for freeing allocate memory - yes, that does make perfect sense.

Reply:
Are you OK with merging this?

Reply:
@tore-espressif I am not sure. 
Probably we need to delete mutex acm->ringbuf_read_mux in case acm->rx_tfbuf allocation error. 
I do not see it in code.

Reply:
@roma-jam @tore-espressif Please check again the new code with additional goto labels.